plugins/org.eclipse.fordiac.ide.application/src/org/eclipse/fordiac/ide/application/marker/resolution/ChangeFileNameMarkerResolution.java

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Felix Schmid
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Felix Schmid - initial API and implementation and/or initial documentation
+ *******************************************************************************/
+package org.eclipse.fordiac.ide.application.marker.resolution;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.fordiac.ide.application.Messages;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeEntry;
+import org.eclipse.fordiac.ide.model.typelibrary.TypeLibraryManager;
+import org.eclipse.fordiac.ide.ui.FordiacLogHelper;
+
+public class ChangeFileNameMarkerResolution extends ChangeNameMarkerResolution {
+
+	public ChangeFileNameMarkerResolution(final IMarker marker) {
+		super(marker);
+	}
+
+	@Override
+	public String getDescription() {
+		return Messages.ChangeFileNameToType_QuickfixDesc;
+	}
+
+	@Override
+	public String getLabel() {
+		return Messages.ChangeFileNameToType_QuickfixLabel;
+	}
+
+	@Override
+	public void run(final IMarker marker) {
+		if (!(marker.getResource() instanceof final IFile file)) {
+			return;
+		}
+		final TypeEntry te = TypeLibraryManager.INSTANCE.getTypeEntryForFile(file);
+		final String newName = te.getTypeName() + "." + marker.getResource().getFileExtension(); //$NON-NLS-1$
```

Comment:
There is a `TypeEntry.getFileExtension()` that returns the expected file extension for that particular type:
```suggestion
		final String newName = te.getTypeName() + "." + te.getFileExtension(); //$NON-NLS-1$
```